depmapomics/ccle.py

```diff
@@ -0,0 +1,1423 @@
+                "format": "TableCSV",
+                "encoding": "utf-8",
+            },
+        ],
+        add_all_existing_files=True,
+        upload_async=False,
+    )
+
+
+def mapBed(file, vcfdir):
+    """map mutations in one vcf file to regions in the guide bed file"""
+
+    guides_bed = pd.read_csv(
+        GUIDESBED, sep="\t", header=None, names=["chrom", "start", "end", "foldchange"]
+    )
+
+    bed = pd.read_csv(
+        vcfdir + file,
+        sep="\t",
+        header=None,
+        names=["chrom", "start", "end", "foldchange"],
+    )
+    bed["foldchange"] = 1
+    name = file.split("/")[-1].split(".")[0].split("_")[1]
+    if len(bed) == 0:
+        return (name, None)
+    val = chip.putInBed(guides_bed, bed, mergetype="sum")
+
+    return (name, val)
+
+
+def generateGermlineMatrix(
+    trackerobj,
+    wesrefworkspace=WESCNWORKSPACE,
+    wgsrefworkspace=WGSWORKSPACE,
+    wesdir=WESVCFDIR,
+    wgsdir=WGSVCFDIR,
+    savedir="output/" + SAMPLESETNAME + "/",
+    bed_location=GUIDESBED,
+    taiga_dataset=TAIGA_CN,
+    vcf_colname="cnn_filtered_vcf",
+    cores=16,
+):
+    """generate profile-level germline mutation matrix for achilles' ancestry correction. VCF files are generated
+    using the CCLE pipeline on terra
+    Args:
+        wesrefworkspace (str, optional): the reference workspace for WES. Defaults to WESMUTWORKSPACE.
+        wgsrefworkspace (str, optional): the reference workspace for WGS. Defaults to WGSWORKSPACE.
+        taiga_dataset (str, optional): taiga folder location. Defaults to TAIGA_CN.
+        wesdir (str, optional): directory where wes vcf files are saved.
+        wgsdir (str, optional): directory where wgs vcf files are saved.
+        savedir (str, optional): directory where output germline matrices are saved.
+        bed_location (str, optional): location of the guides bed file.
+        vcf_colname (str, optional): vcf column name on terra.
+        cores (int, optional): number of cores in parallel processing.
+    
+    Returns:
+        sorted_mat (pd.DataFrame): binary matrix where each row is a region in the guide, and each column corresponds to a profile
+    """
+
+    print("generating germline matrix for wes")
+    h.createFoldersFor(savedir)
+    # load vcfs from WES workspace using dalmatian
+    wm = dm.WorkspaceManager(wesrefworkspace)
+    samp = wm.get_samples()
+    vcfs = samp[vcf_colname]
+    vcfslist = vcfs[~vcfs.isna()].tolist()
+    h.createFoldersFor(wesdir)
+    guides_bed = pd.read_csv(
+        bed_location,
+        sep="\t",
+        header=None,
+        names=["chrom", "start", "end", "foldchange"],
+    )
+    # save vcfs from WES workspace locally,
+    # and run bcftools query to transform vcfs into format needed for subsequent steps
+    # only including regions in the guide bed file
+    cmd = [
+        "gsutil cp "
+        + sam
+        + " "
+        + wesdir
+        + sam.split("/")[-1]
+        + "&& \
+ bcftools index "
```

Comment:
" && bcftools index"

Reply:
we won't need to do gsutil cp and bcftools index in the mutect2 version.

Also, check that the user has bcftools installed maybe

Reply:
Can we test it with the mutect2 version now?